modules/widgets/migrate-to-core/image-widget.php

```diff
@@ -40,13 +79,25 @@ function jetpack_migrate_image_widget() {
 		'conditions' => null,
 	);
 
-	$media_image      = get_option( 'widget_media_image' );
+	$old_widgets = get_option( 'widget_image', array() );
+	$media_image = get_option( 'widget_media_image' );
 	$sidebars_widgets = wp_get_sidebars_widgets();
 
+	// Persist old and current widgets in backup table.
+	jetpack_store_legacy_image_widget_options( 'widget_image', serialize( $old_widgets ) );
+	if ( jetpack_get_legacy_image_widget_option( 'widget_image' ) !== $old_widgets ) {
+		return false;
+	}
+
+	jetpack_store_legacy_image_widget_options( 'sidebars_widgets', serialize( $sidebars_widgets ) );
```

Comment:
I'd argue that `serialize` works just as well here - we know that the data is always an array, and it will always be serialized. But using `maybe_serialize` is OK too, so I changed it.